data_structures/reverse_a_linked_list.cpp

```diff
@@ -0,0 +1,173 @@
+/**
+ * @file
+ * @brief Implementation of reversing a single linked list algorithm.
+ * @details
+ * The linked list is a data structure used for holding a sequence of
+ * values, which can be added, displayed,reversed ,removed.
+ * ### Algorithm
+ * Values can be added by iterating to the end of a list(by following
+ * the pointers) starting from the first link. Whichever link points to null
+ * is considered the last link and is pointed to the new value.
+ *
+ * Linked List can be reversed by using 3 pointers: current, previous, and
+ * next_node; we keep iterating until the last node. Meanwhile, before changing
+ * to the next of current, we store it in the next_node pointer, now we store
+ * the prev pointer in the current of next, this is where the actual reversal
+ * happens. And then we move the prev and current pointers one step forward.
+ * Then the head node is made to point to the last node (prev pointer) after
+ * completion of an iteration.
+ * Graphic Explanation:https://bit.ly/38JLVRd
+ */
+#include <iostream>
+#include <memory>
+#include <cassert> /// for assert
+
+/**
+ * @namespace data_structures
+ * @brief Data Structures algorithms
+ */
+namespace data_structures {
+
+/**
+ * @namespace linked_list
+ * @brief Functions for singly linked list algorithm
+ */
+namespace linked_list {
+/**
+ * A Node class containing a value and pointer to another link
+ */
+class Node
+{
+public:
+int val;     ///< value of the current link
+Node *next;  ///< pointer to the next value on the list
+};
+
+/**
+ * A list class containing a sequence of links
+ */
+class list {
+ private:
+    Node *head{nullptr};  // link before the actual first element
+ public:
+    /**
+     * List constructor. Initializes the first link.
+     */
+    list() {
+        // Initialize the first link
+    }
+
+    void insert(int new_elem);
+    void reverseList();
+    void display();
+    int top();
+    int last();
+};
+
+
+
+/**
+ * function adds new element to the end of the list
+ * @param new_elem to be added to the end of the list
+ */
+void list::insert(int n) {
```

Comment:
[Use specific data type](https://en.cppreference.com/w/cpp/types/integer) 
Maybe int32_t or int64_t accordingly 
```suggestion
void list::insert(int32_t n) {
```